Fix Neo4j unit tests

Because the property `spring.data.neo4j.uri` is set in application.properties, some unit test don't work anymore. If the property is null, they work.

As a workaround I temporarily ignored two unit test classes:
* `MicoApplicationApplicationTests`
* `GitHubCrawlerTests`

**Log:**
[ERROR] testStoreApplication(io.github.ust.mico.core.MicoCoreApplicationTests)  Time elapsed: 3.122 s  <<< ERROR!
org.springframework.transaction.CannotCreateTransactionException: Could not open Neo4j Session for transaction; nested exception is org.neo4j.ogm.exception.ConnectionException: Could not create driver instance
Caused by: org.neo4j.ogm.exception.ConnectionException: Could not create driver instance
Caused by: org.neo4j.driver.v1.exceptions.ServiceUnavailableException: Unable to connect to localhost:7687, ensure the database is running and that there is a working network connection to it.
Caused by: org.neo4j.driver.internal.shaded.io.netty.channel.AbstractChannel$AnnotatedConnectException: Connection refused: localhost/127.0.0.1:7687
Caused by: java.net.ConnectException: Connection refused


Possible solution:
Use e.g. Spring profiles to set different values (or no values) depending on the environment.

--- a/install/kubernetes/mico-core.yaml
+++ b/install/kubernetes/mico-core.yaml
@@ -1,5 +1,16 @@
 ---
 apiVersion: v1
+kind: ConfigMap
+metadata:
+  creationTimestamp: 2017-09-19T04:03:37Z
+  name: mico-core
+  namespace: mico-system
+data:
+  application.properties: |-
+    spring.data.neo4j.uri=bolt://neo4j.mico-system:7687
+    kubernetes.prometheus.uri=http://prometheus.monitoring:9090
+---
+apiVersion: v1
 kind: Service
 metadata:
   name: mico-core
@@ -33,10 +44,5 @@ spec:
       - name: mico-core
         image: ustmico/mico-core
         imagePullPolicy: Always
-        env:
-          - name: NEO4J_BOLT_URI
-            value: 'bolt://neo4j.mico-system:7687'
-          - name: PROMETHEUS_URI
-            value: 'http://prometheus.monitoring:9090'
         ports:
         - containerPort: 8080

--- a/mico-core/pom.xml
+++ b/mico-core/pom.xml
@@ -55,6 +55,11 @@
             <optional>true</optional>
         </dependency>
         <dependency>
+            <groupId>org.springframework.cloud</groupId>
+            <artifactId>spring-cloud-starter-kubernetes-config</artifactId>
+            <version>1.0.0.RELEASE</version>
+        </dependency>
+        <dependency>
             <groupId>org.neo4j</groupId>
             <artifactId>neo4j-ogm-test</artifactId>
             <version>${neo4j-ogm.version}</version>

--- a/mico-core/src/main/resources/application.properties
+++ b/mico-core/src/main/resources/application.properties
@@ -1,8 +1,14 @@
 # Application
-spring.application.name=MICO Core
-server.port = 8080
-# Database
-spring.data.neo4j.uri=${NEO4J_BOLT_URI:bolt://localhost}
+spring.application.name=mico-core
+server.port=8080
+# Database (uncomment if you want to use a real instead of an embedded database)
+#spring.data.neo4j.uri=bolt://localhost
+# Actuator
+management.endpoint.restart.enabled=true
+management.endpoints.web.exposure.include=info,health,loggers,configprops,metrics
+# Spring Cloud Kubernetes
+spring.cloud.kubernetes.config.name=mico-core
+spring.cloud.kubernetes.reload.enabled=false
 # Logging
 logging.level.io.github.ust.mico.core=debug
 # Cors Origins, It is a comma separated list
@@ -17,4 +23,4 @@ kubernetes.build-bot.docker-image-repository-url=docker.io/ustmico
 kubernetes.build-bot.docker-registry-service-account-name=build-bot-dockerhub
 kubernetes.build-bot.kaniko-executor-image-url=gcr.io/kaniko-project/executor
 kubernetes.build-bot.build-timeout=60
-kubernetes.prometheus.uri=${PROMETHEUS_URI:"http://localhost:9090/api/v1/query"}
+kubernetes.prometheus.uri=http://localhost:9090/api/v1/query

--- a/mico-core/src/test/java/io/github/ust/mico/core/GitHubCrawlerTests.java
+++ b/mico-core/src/test/java/io/github/ust/mico/core/GitHubCrawlerTests.java
@@ -1,6 +1,5 @@
 package io.github.ust.mico.core;
 
-import org.junit.Ignore;
 import org.junit.Test;
 import org.junit.runner.RunWith;
 import org.springframework.boot.test.autoconfigure.OverrideAutoConfiguration;
@@ -12,9 +11,6 @@ import static org.junit.Assert.assertEquals;
 
 @RunWith(SpringRunner.class)
 @SpringBootTest
-
-// TODO Only ignored because of Neo4j error
-@Ignore
 public class GitHubCrawlerTests extends Neo4jTestClass {
 
     private static final String REPO_URI_API = "https://api.github.com/repos/octokit/octokit.rb";

--- a/mico-core/src/test/java/io/github/ust/mico/core/MicoCoreApplicationTests.java
+++ b/mico-core/src/test/java/io/github/ust/mico/core/MicoCoreApplicationTests.java
@@ -28,9 +28,6 @@ import io.github.ust.mico.core.persistence.MicoServiceRepository;
 
 @RunWith(SpringRunner.class)
 @SpringBootTest
-
-// TODO Only ignored because of Neo4j error
-@Ignore
 public class MicoCoreApplicationTests extends Neo4jTestClass {
 
     private static final String TEST_SHORT_NAME = "Test";

